src/lib.rs

```diff
@@ -1153,12 +1155,6 @@ mod panic {
 
 struct LoggerGuard(&'static Log);
 
-impl Drop for LoggerGuard {
```

Comment:
Also, I think the load/stores from `STATE` only need `Ordering::Acquire`/`Release` not `SeqCst`. Using `SeqCst` synchronizes with any unrelated `SeqCst` read/writes somewhere else in the program.